src/Host.cpp

```diff
@@ -222,19 +222,85 @@ Host::~Host()
 void Host::saveModules(int sync, bool backup)
 {
     QMapIterator<QString, QStringList> it(modulesToWrite);
-    QStringList modulesToSync;
-    QString dirName = mudlet::getMudletPath(mudlet::moduleBackupsPath);
-    QDir savePath = QDir(dirName);
-    if (!savePath.exists()) {
-        savePath.mkpath(dirName);
+    mModulesToSync.clear();
+    QString savePath = mudlet::getMudletPath(mudlet::moduleBackupsPath);
+    auto savePathDir = QDir(savePath);
+    if (!savePathDir.exists()) {
+        savePathDir.mkpath(savePath);
     }
     while (it.hasNext()) {
         it.next();
         QStringList entry = it.value();
+        QString moduleName = it.key();
         QString filename_xml = entry[0];
-        // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
-        QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+
+        if (backup) {
+            // CHECKME: Consider changing datetime spec to more "sortable" "yyyy-MM-dd#HH-mm-ss" (1 of 6)
+            QString time = QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss");
+            savePathDir.rename(filename_xml, savePath + moduleName + time); //move the old file, use the key (module name) as the file
+        }
+
+        auto writer = new XMLexport(this);
+        writers.insert(filename_xml, writer);
+        writer->writeModuleXML(moduleName, filename_xml);
+
+        if (entry[1].toInt()) {
+            mModulesToSync << moduleName;
+        }
+    }
+    modulesToWrite.clear();
+
+    if (sync) {
+        connect(this, &Host::profileSaveFinished, this, &Host::slot_reloadModules);
+    }
+}
+
+void Host::slot_reloadModules()
+{
+    //synchronize modules across sessions
+    QMap<Host*, TConsole*> activeSessions = mudlet::self()->mConsoleMap;
+    QMapIterator<Host*, TConsole*> sessionIterator(activeSessions);
+    while (sessionIterator.hasNext()) {
+        sessionIterator.next();
+        Host* host = sessionIterator.key();
+        if (host->mHostName == mHostName) {
+            continue;
+        }
+        QMap<QString, int> modulePri = host->mModulePriorities;
+        QMap<int, QStringList> moduleOrder;
+        for (auto it = modulePri.keyBegin(); it != modulePri.keyEnd(); ++it) {
+            moduleOrder[modulePri[*it]].append(*it);
+        }
+        QMapIterator<int, QStringList> it(moduleOrder);
+        while (it.hasNext()) {
+            it.next();
+            QStringList moduleList = it.value();
+            for (int i = 0, total = moduleList.size(); i < total; ++i) {
+                QString moduleName = moduleList[i];
+                if (mModulesToSync.contains(moduleName)) {
+                    host->reloadModule(moduleName);
+                }
+            }
+        }
+    }
+
+    // update the module zips
+    updateModuleZips();
```

Comment:
Shouldn't this be ***before*** the modules get reloaded by the other profiles/Hosts?